invenio_users_resources/services/users/service.py

```diff
@@ -74,7 +76,7 @@ def read(self, identity, id_):
         # resolve and require permission
         user = UserAggregate.get_record(id_)
         if user is None:
-            raise LookupError(f"No user with id '{id_}'.")
+            raise UserNotFoundError(f"User not found '{user}'")
```

Comment:
```suggestion
            raise UserNotFoundError(f"User with if id '{id_}' not found.")
```
because `user` is None in that case...